lab4/provided/.gitignore

```diff
@@ -1,3 +1,4 @@
 cache_simulator
 *.d
 *.o
+.vscode
```

Comment:
Your editor is weird. There's a number of places where it doesn't put a newline at the end of the file. No need to fix, but you might want to change your editor settings for the future.